File: packages/java/parser-jvm-plugin-transfertypes/src/test/java/com/vaadin/hilla/parser/plugins/transfertypes/signals/SignalEndpoint.java
package com.vaadin.hilla.parser.plugins.transfertypes.signals;

import com.vaadin.hilla.signals.ListSignal;
import com.vaadin.hilla.signals.NumberSignal;
import com.vaadin.hilla.signals.Signal;
import com.vaadin.hilla.signals.ValueSignal;

@Endpoint
public class SignalEndpoint {
    public Signal<String> getStringSignal() {
        return null;
    }

    public ValueSignal<String> getStringValueSignal() {
        return null;
    }

    public NumberSignal getNumberSignal() {
        return null;
    }

    public ListSignal<String> getStringListSignal() {
        return null;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/transfertypes/signals/snapshots/SignalEndpoint.ts
import { EndpointRequestInit } from "@vaadin/hilla-frontend";
import { ListSignal, NumberSignal, Signal, SignalMethodOptions, ValueSignal } from "@vaadin/hilla-react-signals";
import client from "./connect-client.default.js";
function getNumberSignal(): NumberSignal | undefined { return new NumberSignal(0, {
    client: client,
    endpoint: "SignalEndpoint",
    method: "getNumberSignal"
}); }
function getStringListSignal(): ListSignal<string | undefined> | undefined { return new ListSignal({
    client: client,
    endpoint: "SignalEndpoint",
    method: "getStringListSignal"
}); }
async function getStringSignal(init?: EndpointRequestInit): Promise<Signal<string | undefined> | undefined> { return client.call("SignalEndpoint", "getStringSignal", {}, init); }
function getStringValueSignal(options?: SignalMethodOptions<string | undefined>): ValueSignal<string | undefined> | undefined { return new ValueSignal(options?.defaultValue, {
    client: client,
    endpoint: "SignalEndpoint",
    method: "getStringValueSignal"
}); }
export { getNumberSignal, getStringListSignal, getStringSignal, getStringValueSignal };
